code/modules/reagents/reagent_containers/cups/bottle.dm

```diff
@@ -457,3 +457,93 @@
 	name = "bottle of caramel"
 	desc = "A bottle containing caramalized sugar, also known as caramel. Do not lick."
 	list_reagents = list(/datum/reagent/consumable/caramel = 30)
+
+/*
+ *	Syrup bottles, basically a unspillable cup that transfers reagents upon clicking on it with a cup
+ */
+
+/obj/item/reagent_containers/cup/bottle/syrup_bottle
+	name = "syrup bottle"
+	desc = "A bottle with a syrup pump to dispense the delicious substance directly into your coffee cup."
+	icon = 'icons/obj/food/containers.dmi'
+	icon_state = "syrup"
+	fill_icon_state = "syrup"
+	fill_icon_thresholds = list(0, 20, 40, 60, 80, 100)
+	possible_transfer_amounts = list(5, 10)
+	volume = 50
+	amount_per_transfer_from_this = 5
+	spillable = FALSE
+	///variable to tell if the bottle can be refilled
+	var/cap_on = TRUE
+
+/obj/item/reagent_containers/cup/bottle/syrup_bottle/examine(mob/user)
+	. = ..()
+	. += span_notice("Alt-click to toggle the pump cap.")
+	return
+
+//when you attack the syrup bottle with a container it refills it
+/obj/item/reagent_containers/cup/bottle/syrup_bottle/attackby(obj/item/attacking_item, mob/user, params)
+	//SHOULD_CALL_PARENT(FALSE)
+
+	if(!cap_on)
+		.=..()
+		return
+
+	if(!check_allowed_items(attacking_item,target_self=1))
+		return
+
+	if(attacking_item.is_refillable())
+		if(!reagents.total_volume)
+			to_chat(user, span_warning("[src] is empty!"))
+			return TRUE
+
+		if(attacking_item.reagents.holder_full())
+			to_chat(user, span_warning("[attacking_item] is full."))
+			return TRUE
+
+		var/trans = reagents.trans_to(attacking_item, amount_per_transfer_from_this, transfered_by = user)
```

Comment:
Full variable name

Reply:
Addressed in [fa2fd29](https://github.com/tgstation/tgstation/pull/70991/commits/fa2fd2913571a945d618ce4441735d29692c5e0c)